src/mesh.cpp

```diff
@@ -3306,16 +3327,56 @@ void LibMesh::initialize()
   // contiguous in ID space, so we need to map from bin indices (defined over
   // active elements) to global dof ids
   if (adaptive_) {
+    //std::cout<<"maping eveything again\n";
     bin_to_elem_map_.reserve(m_->n_active_elem());
     elem_to_bin_map_.resize(m_->n_elem(), -1);
+
+    //clear the previous history
+    clustering_element_mapping_.clear();
```

Comment:
Probably don't need to clear this - it will be fully initialized each time, but may be more efficient/necessary to reserve size like they do for `bin_to_elem_map_`.